### ISSUE TITLE
Visual Code 1.18 unusable when file contains many breakpoints

### ISSUE THREAD

**Issue by aiafrasinei** [Microsoft/vscode#38076]
- VSCode Version: Code 1.18.0 (dcee2202709a4f223185514b9275aa4229841aa7, 2017-11-08T12:44:48.751Z)
- OS Version: Linux x64 4.4.0-98-generic
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
cpptools|ms-|0.14.2

---

Steps to Reproduce:

1.Have a c++ file about 10k lines.Copy paste code and editor is blocked.
Seems to be happening random dont have more details not doing anything special.
This is horrendous cant use the editor basically.

![image](https://user-images.githubusercontent.com/5115332/32668773-332849aa-c647-11e7-94b6-4919f7e36093.png)

I use cpptools but that didnt seem to be the problem, i disable it and same story.

Reproduces without extensions: Yes

**Comment by aiafrasinei** [comment]
Something i noticed. I have a lot of breakpoints i didnt add to this file.

**Comment by bpasero** [comment]
I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line (NOT the integrated terminal in Code), execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by aiafrasinei** [comment]
There were lots of breakpoints in the file i didnt add by hand. 
Found a workaround. I removed all the breakpoints and all seems ok now.

Maybe is related to this but im not sure.

**Comment by thigh** [comment]
I just experienced the same problem with some .js files that had a ton of breakpoints that I did not create.  I deleted all of the breakpoints and now everything is performing well.

From what I can gather it looks like after installing the 1.18.0 update, all (most?) of the files that had at least some pre-existing breakpoints suddenly had many breakpoints added.  In some cases the breakpoints seemed invalid as the particular line would not support a breakpoint in VSC under normal circumstances.  Every file with the extra breakpoints would suffer from performance when performing very specific actions that caused the file to update (e.g. writing on the same line was fine but attempting to delete or add a line would cause the editor to freeze for several seconds).
